go.mod

```diff
@@ -1,9 +1,11 @@
 module github.com/goto/salt
 
-go 1.16
+go 1.21
+
+toolchain go1.21.8
```

Comment:
do we need this?

Reply:
on go mod tidy -> automatically gets updated to go 1.21, 
when trying to enforce 1.16 usage, some packages of otel are not present in go 1.16

Reply:
removed